### PR TITLE
chore(core-flows): adjust inventory when return is received

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
* Adjust inventory when receiving item return
* Fix stock check adding to cart with updated quantity
* Fix Order module indexes

### CODE REVIEW

File: integration-tests/http/__tests__/returns/returns.spec.ts [hunk 1]
```diff
@@ -14,11 +19,30 @@ medusaIntegrationTestRunner({
...
     beforeEach(async () => {
       const container = getContainer()
       await createAdminUser(dbConnection, adminHeaders, container)
 
+      const inventoryModule = container.resolve(
+        ModuleRegistrationName.INVENTORY
+      )
+      const productModule = container.resolve(ModuleRegistrationName.PRODUCT)
+
+      const [product] = await productModule.createProducts([
```
Comment (olivermrbl) [inline integration-tests/http/__tests__/returns/returns.spec.ts]: nit(non-blocking): Use `POST /admin/products`

File: integration-tests/http/__tests__/returns/returns.spec.ts [hunk 2]
```diff
@@ -123,6 +148,50 @@ medusaIntegrationTestRunner({
         )
       ).data.fulfillment_set
 
+      inventoryItem = await inventoryModule.createInventoryItems({
```
Comment (olivermrbl) [inline integration-tests/http/__tests__/returns/returns.spec.ts]: nit(non-blocking): Use API endpoints

File: integration-tests/modules/__tests__/order/workflows/create-fulfillment.spec.ts [hunk 3]
```diff
@@ -351,6 +352,9 @@ medusaIntegrationTestRunner({
         )
 
         const order = await createOrderFixture({ container, product, location })
+        const itemWithInventory = order.items!.find(
+          (o) => o.variant_sku === "test-variant"
```
Comment (olivermrbl) [inline integration-tests/modules/__tests__/order/workflows/create-fulfillment.spec.ts]: nit(non-blocking): Extract "test-variant" to const

File: packages/modules/order/src/migrations/Migration20240604100512.ts [hunk 4]
```diff
@@ -32,17 +32,17 @@ export class Migration20240604100512 extends Migration {
     CREATE INDEX IF NOT EXISTS "IDX_order_transaction_return_id" ON "order_transaction" (
```
Comment (olivermrbl) [inline packages/modules/order/src/migrations/Migration20240604100512.ts:32]: question: nice catch, should the changes to indexes come in a separate migration?

File: packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts [hunk 5]
```diff
@@ -40,13 +48,95 @@ const validationStep = createStep(
...
+          variant_id: inventory_items.variant_id,
+          inventory_item_id: inventoryItemId,
+          required_quantity: inventory_items.required_quantity,
+        })
+      }
+    }
+  )
+
+  if (hasManagedInventory && !hasStockLocation) {
+    // TODO: how to bypass this error?
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts]: question: This looks OK at first glance. What were your thoughts here?

File: packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts [hunk 6]
```diff
@@ -40,13 +48,95 @@ const validationStep = createStep(
   }
 )
 
+function prepareInventoryUpdate({ orderReturn, returnedQuantityMap }) {
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts:52]: nit: Can I get you to add a comment describing the logic in this function? I think it's a bit hard to follow, so some context might help for the next reader.

File: packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts [hunk 7]
```diff
@@ -81,45 +171,84 @@ export const confirmReturnReceiveWorkflow = createWorkflow(
...
+              )?.item?.variant_id
 
-        itemUpdates[itemId] = {
-          id: itemMap[itemId],
-          received_quantity: act.details!.quantity,
+              if (!variantId) {
+                return
+              }
+
+              const cur = returnedQuantityMap[variantId] ?? 0
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts]: nit(non-blocking): Rename `cur` to something more sensible

File: packages/modules/order/src/migrations/Migration20240219102530.ts [hunk 8]
```diff
@@ -115,37 +115,37 @@ export class Migration20240219102530 extends Migration {
       CREATE INDEX IF NOT EXISTS "IDX_order_display_id" ON "order" (
```
Comment (adrien2p) [inline packages/modules/order/src/migrations/Migration20240219102530.ts:115]: Q: when handling migration changes, do you use the command or do you do it manually? Cause if the snapshot are not up to date we will never be able to use the commands again :D so many at some point we need to be sure that all modules snapshots are up to data wdyt?

Comment (carlos-r-l-rodrigues) [reply packages/modules/order/src/migrations/Migration20240219102530.ts:115]: I always do it manually.
yes, I think we have to have just 1 single initial migration file per module with the final snapshot updated before releasing them in v2.

File: packages/modules/order/src/migrations/Migration20240604100512.ts (hunk 4 above)
Comment (carlos-r-l-rodrigues) [reply packages/modules/order/src/migrations/Migration20240604100512.ts:32]: These are fine to live here because it will only impact performance and not the functionality.
And I believe before releasing we should have just 1 single migration file for each module.

File: packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts (hunk 5 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts]: Just thinking if someone get stuck because of this condition, if they could somehow bypass it. But maybe they need to update the location of the return and then confirm this.

File: packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts (hunk 6 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts:52]: I added a couple of comments along the function.
I hope that helps a bit

Comment (olivermrbl) [reply packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts:52]: This is great, thanks!

**Review by olivermrbl — APPROVED** [review]
